python/adelphi/adelphi/nb.py

```diff
@@ -47,11 +47,14 @@ def build_select_statements(keyspace, table):
 
 
 def build_insert_statements(keyspace, table):
+    # Note that both the sequence of column names and column bindings are built off of
+    # the same base sequence (cols below) in order to make sure column names and binding
+    # names line up in the generated CQL.  Order is pretty important here.
     cols = table.columns.values()
-    primary_keys = set(table.primary_key)
+    primary_keys = set([c.name for c in table.primary_key])
```

Comment:
The root cause of the problem with the integration tests.

We built up a set of ColumnMetadata instances returned from the driver and then traverse the list of known ColumnMetadata instances for a table, checking each to see if it's in that set and generating different nosqlbench seq/dist names accordingly.  But for some reason an input ColumnMetadata instance wasn't being identified as a member of the set even though it very definitely was in the set (something I validated two or three different ways).  It works fine to use the column _name_ rather than the column instance directly so I just wound up doing that but... I mean... :huh: